File: app/src/main/java/jp/bellware/echo/setting/SettingActivity.java
package jp.bellware.echo.setting;

import android.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;

import jp.bellware.echo.R;
import jp.bellware.echo.analytics.AnalyticsHandler;

public class SettingActivity extends AppCompatActivity {

    private AnalyticsHandler ah = new AnalyticsHandler();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);
        //ツールバーの設定
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        //フラグメントの追加
        FragmentTransaction ft =getFragmentManager().beginTransaction();
        ft.replace(R.id.preference_frame,new SettingFragment());
        ft.commit();
        //Analytics
        ah.onCreate(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        ah.onResume();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if(item.getItemId() == android.R.id.home){
            finish();
            return true;
        }
        return false;
    }
}

File: app/src/main/java/jp/bellware/echo/setting/SettingActivity.kt
package jp.bellware.echo.setting

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import jp.bellware.echo.R
import jp.bellware.echo.analytics.AnalyticsHandler
import kotlinx.android.synthetic.main.activity_setting.*

/**
 * 設定画面
 */
class SettingActivity : AppCompatActivity() {

    private val ah = AnalyticsHandler()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)
        //ツールバーの設定
        setSupportActionBar(toolbar)
        val lsab = supportActionBar
        lsab?.setHomeButtonEnabled(true)
        lsab?.setDisplayHomeAsUpEnabled(true)
        //フラグメントの追加
        val ft = supportFragmentManager.beginTransaction()
        ft.replace(R.id.preference_frame, SettingFragment())
        ft.commit()
        //Analytics
        ah.onCreate(this)
    }

    override fun onResume() {
        super.onResume()
        ah.onResume()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return false
    }
}
